adr/ADR-8.md

```diff
@@ -216,7 +216,11 @@ Writing a key to the bucket is a basic JetStream request.
 The KV key `auth.username` in the `CONFIGURATION` bucket is written sent, using a request, to `$KV.CONFIGURATION.auth.username`.
 
 To implement the feature that would accept a write only if the sequence of the current value of a key has a specific sequence
-we use the new `Nats-Expected-Last-Subject-Sequence` header.
+we use the new `Nats-Expected-Last-Subject-Sequence` header. Since a recent server version the special value `0` for this header
+would indicate that the message should only be accepted if it's the first message on a subject. This is purge aware, ie. if 
+a value is in and the subject is purged again a `0` value will be accepted.
+
+This can be implemented as a `PutOption` ie. `Put("x.y", []byte("hello world"), OnlyIfLastValueSequence(10))`.
```

Comment:
Please suggest better name than `OnlyIfLastValueSequence`

Reply:
I suppose this is intended to solve the create vs put problem but does not because what if a key's last operation is delete, a create should work. Maybe a purge should happen first. Besides the create scenario, I can't see a user using this at all. Maybe we just have to add create to the api

Reply:
There are many things people do with KV around service discovery where either a specific thing should be standby/fiirst, where without updates expires should happen, where you should be able to iterate members in a set etc. 

All weird and wonderful things people do with KV (see Consul service registry), so you might not have used it in that case and indeed for the typical put some config stuff in a bucket its entirely useless, but being present in the API mean we can fix other problems too

There's a whole other world of things people build with KV that I am working on - service registries is one of the major reasons people run consul.

Reply:
You could just name it to match the header name, i.e. `ExpectedLastSequence`